### PR TITLE
LKE-6923 feat(spaces): Add getSpacesList API

### PR DESCRIPTION
**Pull request by aminerhanemi**

### CODE REVIEW

File: src/api/spaces/index.ts [hunk 1]
```diff
@@ -19,4 +19,15 @@ export class SpacesAPI extends Request {
       params: params
     });
   }
+
+  /**
+   * Get spaces list.
+   */
+  public getSpacesList(this: Request<ISpace[]>) {
```
Comment (gmasclet) [inline src/api/spaces/index.ts:26]: I suggest simplifying the method name to `getSpaces` and adding the source key parameter:

```suggestion
  public getSpaces(this: Request<ISpace[]>, params?: IDataSourceParams) {
```

File: src/api/spaces/index.ts [hunk 2]
```diff
@@ -19,4 +19,15 @@ export class SpacesAPI extends Request {
       params: params
     });
   }
+
+  /**
+   * Get spaces list.
```
Comment (gmasclet) [inline src/api/spaces/index.ts:24]: nit:

```suggestion
   * List spaces.
```

File: src/api/spaces/index.ts (hunk 1 above)
Comment (aminerhanemi) [reply src/api/spaces/index.ts:33]: what is the need of the sourcekey params ?

File: src/api/spaces/index.ts (hunk 2 above)
Comment (aminerhanemi) [reply src/api/spaces/index.ts:31]: I like starting with a verb to show it is an action

**Review by gmasclet — APPROVED** [review]
LGTM :+1:

**Pull request merged**